### PR TITLE
Admin Page with Scheduling, Editing, Deletion, Refreshes

### PR DESCRIPTION
**Pull request by dlarocque**
- Routing to Admin page (if signed in) using the [React Router](https://reactrouter.com/en/main)
- Displaying all release metadata in table format (not releases, changes, checks, since those are not editable)
- Each release can be interacted with using buttons for sync, edit, delete
- Releases can be added one at a time by filling out the add release form
- Error and success messages are displayed in a [snackbar](https://mui.com/material-ui/react-snackbar/)

Future Improvements:
- Stack trace on errors rather than opaque error messages
- Batch add release (from CSV)
- Restrict the /admin page to only be accessible to signed in users (the nav button isn't currently displayed, but a user that is signed out can still manually access that URL) using [ProtectedRoute](https://www.robinwieruch.de/react-router-private-routes/)

The live version is not out yet since this relies on functions, which I don't have permission to deploy yet. I can share some screenshots in the meantime

### CODE REVIEW

File: firebase-android-release-dashboard/src/components/AdminPage/AddReleaseForm/AddReleaseForm.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    releaseDate: "",
+  });
+
+  /**
+   * Check if the form is valid every time the form data changes.
+   * The form is valid if all fields are filled out.
+   */
+  useEffect(() => {
+    const isFormValid = Object.values(formData).every((value) => value !== "");
+    setFormValid(isFormValid);
```
Comment (daymxn) [inline firebase-android-release-dashboard/src/components/AdminPage/AddReleaseForm/AddReleaseForm.js:42]: Great usage of a variable to portray purpose!

File: firebase-android-release-dashboard/src/components/AdminPage/ReleaseRow/ReleaseRow.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      setEditedRelease({
+        ...editedRelease,
+        [event.target.name]: new Date(event.target.value),
+      });
+    } else {
+      setEditedRelease({
+        ...editedRelease,
+        [event.target.name]: event.target.value,
+      });
+    }
```
Comment (daymxn) [inline firebase-android-release-dashboard/src/components/AdminPage/ReleaseRow/ReleaseRow.js:89]: You could reduce this to:
```js

const handleChange = (event) => {
    const isDateType = ["releaseDate", "codeFreezeDate"].includes(event.target.name);
    
     setEditedRelease({
        ...editedRelease,
        [event.target.name]: isDateType ? new Date(event.target.value) : event.target.value
     });
}
```

Or something of the like

**Review by daymxn — APPROVED** [review]
Great job! Left a couple comments, but lgtm:)

File: firebase-android-release-dashboard/src/components/AdminPage/ReleaseRow/ReleaseRow.js (hunk 2 above)
Comment (dlarocque) [reply firebase-android-release-dashboard/src/components/AdminPage/ReleaseRow/ReleaseRow.js]: That's much cleaner, thanks!

**Review by davidmotson — APPROVED** [review]

**Pull request merged**